corehq/apps/locations/forms.py

```diff
@@ -195,10 +195,14 @@ def clean_parent_id(self):
 
     def clean_name(self):
         name = self.cleaned_data['name']
+        parent = self.cleaned_data.get('parent')
 
         if self.strict:
-            siblings = self.location.siblings(self.cleaned_data.get('parent'))
-            if name in [loc.name for loc in siblings]:
+            if (SQLLocation.objects.filter(domain=self.location.domain,
+                                           parent=parent,
+                                           name=name)
+                                   .exclude(location_id=self.location._id)
```

Comment:
I certainly hope not!
This is getting all locations which list `parent` as the `parent`, but excludes the current location.  The idea is, when setting the parent, make sure that there are no other children of that parent with the same name.